### PR TITLE
feat: add sharpness unit

### PR DESCRIPTION
**Pull request by axelstr**
Sharpness is a numerical measurement of angular change per physical
length.  And example is the sharpness of a curve.

### CODE REVIEW

**Review by john-liu-einride — APPROVED** [review]
:hocho:

File: sharpness.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	magnitude, err := strconv.ParseFloat(strNoUnit, 64)
+	if err != nil {
+		return fmt.Errorf("unmarshal sharpness: parse '%s': %w", s, err)
+	}
+	*s = Sharpness(magnitude)
+	return nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler.
```
Comment (joaoeinride) [inline sharpness.go:50]: would be good to have this as a type assert like

```suggestion
var _ encoding.TextUnmarshaler = (*Sharpness)(nil)
```

to statically assert it. same  for `encoding.TextUnmarhsaler`

wdyt?

Comment (axelstr) [reply sharpness.go:53]: I added it for all types as a separate commit. Thanks for the suggestion!

Comment (joaoeinride) [reply sharpness.go:53]: I saw we use the other format I couldn't get that to work

**Pull request merged**